tests/python/pants_test/util/test_objects.py

```diff
@@ -73,14 +73,16 @@ def test_validate(self):
     superclasses_of_a_or_b = SuperclassesOf(self.A, self.B)
     self.assertEqual(self.A(), superclasses_of_a_or_b.validate_satisfied_by(self.A()))
     self.assertEqual(self.B(), superclasses_of_a_or_b.validate_satisfied_by(self.B()))
-    with self.assertRaisesRegexp(TypeConstraintError,
-                                 re.escape("value C() (with type 'C') must satisfy this type constraint: SuperclassesOf(A or B).")):
+    with self.assertRaisesWithMessageContaining(
+        TypeConstraintError,
+        "value C() (with type 'C') must satisfy this type constraint: SuperclassesOf(A or B)."):
```

Comment:
Any reason this isn't `exact`?

Reply:
Nope! Fixed by changing the default to `substring_match=False`!